app/src/main/java/com/devshish/internship/adapter/ItemSongAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.devshish.internship.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.devshish.internship.databinding.ItemSongBinding
+import com.devshish.internship.model.Song
+
+class ItemSongAdapter : RecyclerView.Adapter<ItemSongAdapter.SongViewHolder>() {
+
+    private var songs = emptyList<Song>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SongViewHolder {
+        val binding = ItemSongBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return SongViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: SongViewHolder, position: Int) {
+        val curSong = songs[position]
+        holder.bind(curSong)
+    }
+
+    override fun getItemCount() = songs.size
+
+    inner class SongViewHolder(
```

Comment:
inner modifier is not best practice on ViewHolder - also when inner features is not used

Reply:
1. remove inner modifier
2. move ViewHolder and DiffCallback to another files
3. Add private val listener: (Song) -> Unit to constructor of view holder